src/main/java/com/aws/greengrass/clientdevices/auth/PermissionEvaluationUtils.java

```diff
@@ -130,12 +131,9 @@ private boolean compareResource(Resource requestResource, String policyResource)
             return true;
         }
 
-        if (String.format(SERVICE_RESOURCE_FORMAT, requestResource.getService(), requestResource.getResourceType(),
-                ANY_REGEX).equals(policyResource)) {
-            return true;
-        }
-
-        return ANY_REGEX.equals(policyResource);
+        WildcardTrie trie = new WildcardTrie();
```

Comment:
Is it feasible to build the trie one time instead of doing it every lookup?

Reply:
yeah, although the tries are unique to a session & resource--caching them would be a little complicated so could explore in a separate change